tickets/src/components/content/purchase/PayButton.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+
+import withConfig from '../../../utils/withConfig'
+import UnlockPropTypes from '../../../propTypes'
+
+export const PayButton = ({ transaction, purchaseKey, config }) => {
+  const { requiredConfirmations } = config
+  if (!transaction || !transaction.status) {
+    return <Pay onClick={purchaseKey}>Pay &amp; Register for This Event</Pay>
+  } else if (
+    transaction &&
+    ['submitted', 'pending'].indexOf(transaction.status) > -1
+  ) {
+    return <PayInfo>Payment Sent ...</PayInfo>
+  } else if (
+    transaction &&
+    transaction.status === 'mined' &&
+    transaction.confirmations < requiredConfirmations
+  ) {
+    return (
+      <PayInfo>
+        Confirming Payment ({transaction.confirmations} /{' '}
+        {requiredConfirmations})
+      </PayInfo>
+    )
+  } else if (transaction && transaction.status === 'mined') {
+    return <PayInfo>Confirmed</PayInfo>
+  }
```

Comment:
How would you feel about adding a fall through `else` here to ensure that even in the worse scenario we will not have a UI in an  unknown state?